sweep/sweeper.go

```diff
@@ -1466,6 +1485,61 @@ func (s *UtxoSweeper) handleNewInput(input *sweepInputMessage) {
 	pi.ntfnRegCancel = cancel
 }
 
+// attachAvailableRBFInfo queries the mempool to see whether the given input
+// has already been spent. If so, it will query the sweeper store to fetch the
+// fee info of the spending transction, hence preparing for possible RBF.
+func (s *UtxoSweeper) attachAvailableRBFInfo(pi *pendingInput) *pendingInput {
```

Comment:
What if instead of

```go
func (*UtxoSweeper) attachAvailableRBFInfo(*pendingInput) *pendingInput {...}
```

we instead made this

```go
func (*UtxoSweeper) findAvailableRBFInfo(pendingInput) Option[RBFInfo] {...}
```

And do the attachment at the call site if it is found? I think that would fully solve my API concerns and clean things up a bit as far as responsibility goes. wdyt?

Reply:
Glad that you mentioned it - it's now changed, with a new name, to emphasize it's doing two things - first decide the state, then the rbf info. I think we can end up in a situation where we have a published input but not found in the store if a third party sweeps it.